pkg/helpers/conditions/conditions.go

```diff
@@ -39,6 +39,10 @@ func PodContainerRunning(containerName string, coreClient corev1client.CoreV1Int
 						}
 					}
 				}
+				// until the pod is marked as running, we we have to loop
+				if t.Status.Phase == corev1.PodPending {
+					return false, nil
+				}
```

Comment:
should be this (i should have separated Running from Pending in previous PR)? 
```shell
switch t.Status.Phase {
			case corev1.PodRunning:
			case corev1.PodPending:
				for _, s := range t.Status.ContainerStatuses {
					if s.State.Waiting != nil {
						// Return error here if pod is pending and container status indicates a failure
						// otherwise, user would have to wait the timeout period (15 min)
						// for pod to exit.
						if s.State.Waiting.Reason == "CreateContainerError" || s.State.Waiting.Reason == "ImagePullBackOff" {
							return false, fmt.Errorf(s.State.Waiting.Message)
						}
					}
				}
				return false, nil
```

Reply:
I'm not sure whether that loop is safe.  If *any* container is waiting doesn't mean your current container can't start.  It's only your container or a previous init container.  And you can get CreateContainerError and ImagePullBackoff very normally.

This might need to be a print instead `Pod is waiting for image pull for 15s` (which is why this loop may not be in the right place)

Reply:
I'm inclined to say this is a general-purpose helper that should only wait for Running, but we need a smarter one inside debug to prevent the problems that the fixes @sallyom added. Iow. when you have `CreateContainerError` or `ImagePullBackoff` or any other error that I'd expect debug to return early.

Reply:
Note that ImagePullBackoff can happen in normal operation and eventually succeed (such as if you create a second and debug pod at the same time).

Reply:
Yeah, but the idea here was to provide fast feedback to `oc debug` author, instead of him going to figure out what has happened only after waiting for indefinite amount of time.